<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">

    <arg name="world_name" value="$(find gazebo_kinect_training)/worlds/gazebo_kitchen.world"/>
  </include>

  <arg name="urdf_file" default="$(find xacro)/xacro.py '$(find gazebo_kinect_training)/urdf/kinect_floating.urdf.xacro'" />
  <param name="robot_description" command="$(arg urdf_file)" />

  
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -x 1.25 -z 0.3  -Y 3.14159 -urdf -model kinect" />

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>

  <!-- Fake laser -->
  <node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
        args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
    <param name="scan_height" value="10"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.45"/>
    <remap from="image" to="/camera/depth/image_raw"/>
    <remap from="scan" to="/scan"/>
  </node>
</launch>
